src/plugins/autologin/MQ2AutoLogin.cpp

```diff
@@ -45,6 +45,96 @@ uint64_t ReenableTime = 0;
 int Level = -1;
 int Class = -1;
 
+class MQ2AutoLoginType* pAutoLoginType = nullptr;
+
+class MQ2AutoLoginType : public MQ2Type
+{
+public:
+	std::optional<ProfileRecord> m_record;
+	enum class AutoLoginMembers
+	{
+		HotKey,
+		Server,
+		Character,
+		Profile,
+		Account,
+		Class,
+		Level
+	};
+
+	MQ2AutoLoginType() : MQ2Type("AutoLogin")
+	{
+		ScopedTypeMember(AutoLoginMembers, HotKey);
+		ScopedTypeMember(AutoLoginMembers, Server);
+		ScopedTypeMember(AutoLoginMembers, Character);
+		ScopedTypeMember(AutoLoginMembers, Profile);
+		ScopedTypeMember(AutoLoginMembers, Account);
+		ScopedTypeMember(AutoLoginMembers, Class);
+		ScopedTypeMember(AutoLoginMembers, Level);
+	}
+
+	virtual bool GetMember(MQVarPtr VarPtr, const char* Member, char* Index, MQTypeVar& Dest) override
+	{
+		MQTypeMember* pMember = MQ2AutoLoginType::FindMember(Member);
+		if (!pMember)
+			return false;
+
+		switch ((AutoLoginMembers)pMember->ID)
+		{
+		case AutoLoginMembers::HotKey:
+			strcpy_s(DataTypeTemp, m_record->hotkey.data());
```

Comment:
Oh, almost missed this -- since m_record is an `optional`, it could be unset. You should check for that and bail at the top of GetMember

Reply:
Added check before switch.